### DISCUSSION TITLE
tooltip specific for different layers

### DISCUSSION THREAD

**Discussion by cornhundred** [visgl/deck.gl#6512] [Q&A] [answered]
Hi, is there a way to have a custom tooltip for specific layers? For instance I would like to have the tooltip show the name of one layer (polygon layer, `object.name`) and the category in addition to the name of a different layer (scatterplot layer, `object.leiden`). But I'm unclear if I can check which layer the object is from before I define the HTML of the tooltip. 

```
  var deckgl = new deck.DeckGL({
    container,
    views:[view],
    initialViewState: initial_view_state,
    getTooltip: ({object}) => object && {
      html: `${object.name}`,
      style: {
        fontSize: '0.8em',
        padding: '5px',
      }
    });

```

**Comment by cornhundred** [comment]
For the time being I'll just make a new `object.tooltip` attribute for the data in each layer so that I can have custom data show up in the tooltip.

**Comment by Pessimistress** [comment, marked as answer]
[getTooltip](https://deck.gl/docs/api-reference/core/deck#gettooltip) is passed a `pickingInfo` object, which contains the layer being picked.
